File: src/main/java/io/github/dqualizer/dqapi/services/dam/domainstory/WorkObjectService.java
package io.github.dqualizer.dqapi.services.dam.domainstory;

import io.github.dqualizer.dqapi.exceptions.NotFoundException;
import io.github.dqualizer.dqapi.repositories.dam.domainstory.WorkObjectRepository;
import io.github.dqualizer.dqlang.types.dam.domainstory.WorkObject;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class WorkObjectService {
    @Autowired
    private WorkObjectRepository repository;

    public List<WorkObject> readAll() {
        return repository.findAll();
    }

    public WorkObject readById(String id) {
        return repository.findById(id).orElseThrow(() -> new NotFoundException("Could not find WorkObject with id: " + id + "."));
    }

    public WorkObject create(WorkObject entity) {
        return null; //TODO
    }


}

File: src/main/kotlin/io/github/dqualizer/dqapi/services/dam/domainstory/WorkObjectService.kt
package io.github.dqualizer.dqapi.services.dam.domainstory

import io.github.dqualizer.dqapi.exceptions.NotFoundException
import io.github.dqualizer.dqapi.repositories.dam.domainstory.WorkObjectRepository
import io.github.dqualizer.dqlang.types.dam.domainstory.WorkObject
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class WorkObjectService {
  @Autowired
  lateinit var repository: WorkObjectRepository

  fun readAll(): MutableList<WorkObject> {
    return repository.findAll()
  }

  fun readById(id: String): WorkObject? {
    return repository.findById(id).orElseThrow { NotFoundException("Could not find WorkObject with id: $id.") }
  }

  fun create(entity: WorkObject?): WorkObject? {
    TODO()
  }
}
